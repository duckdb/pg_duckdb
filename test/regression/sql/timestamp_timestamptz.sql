----------------------------------------
-- Timestamp tests
----------------------------------------
-- Test +/- inf values
CREATE TABLE t(a TIMESTAMP, b TEXT);
INSERT INTO t VALUES('Infinity','Positive INF'), ('-Infinity','Negative INF');

-- PG Execution
SELECT * from t;
SELECT isfinite(a),b FROM t;

set duckdb.force_execution = true;
-- DuckDB execution
SELECT * from t;
SELECT isfinite(a),b FROM t;

-- Cleanup
set duckdb.force_execution = false;
DROP TABLE t;

SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00'::timestamp_s as timestamp_s $$);
SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59'::timestamp_s as timestamp_s $$);  -- out of range
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59'::timestamp_s as timestamp_s $$);
SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00'::timestamp_s as timestamp_s $$);  -- out of range

SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00.000000'::timestamp as timestamp $$);
SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59.999999'::timestamp as timestamp $$);  -- out of range
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59.999999'::timestamp as timestamp $$);
SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00.000000'::timestamp as timestamp $$);  -- out of range

SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00.000'::timestamp_ms as timestamp_ms $$);
SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59.999'::timestamp_ms as timestamp_ms $$);  -- out of range
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59.999'::timestamp_ms as timestamp_ms $$);
SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00.000'::timestamp_ms as timestamp_ms $$);  -- out of range
----------------------------------------
-- TimestampTz tests
----------------------------------------
-- Test +/- inf valuestz
CREATE TABLE t(a TIMESTAMPTZ, b TEXT);
INSERT INTO t VALUES('Infinity','Positive INF'), ('-Infinity','Negative INF');

-- PG Execution
SELECT * from t;
SELECT isfinite(a), b FROM t;

set duckdb.force_execution = true;
-- DuckDB execution
SELECT * from t;
SELECT isfinite(a), b FROM t;

-- Cleanup
set duckdb.force_execution = false;
DROP TABLE t;

SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00+00'::timestamptz as timestamptz $$);
SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59+00'::timestamptz as timestamptz $$);  -- out of range
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59+00'::timestamptz as timestamptz $$);
SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00+00'::timestamptz as timestamptz $$);  -- out of range
