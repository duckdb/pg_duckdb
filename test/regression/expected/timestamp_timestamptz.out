----------------------------------------
-- Timestamp tests
----------------------------------------
-- Test +/- inf values
CREATE TABLE t(a TIMESTAMP, b TEXT);
INSERT INTO t VALUES('Infinity','Positive INF'), ('-Infinity','Negative INF');
-- PG Execution
SELECT * from t;
     a     |      b       
-----------+--------------
 infinity  | Positive INF
 -infinity | Negative INF
(2 rows)

SELECT isfinite(a),b FROM t;
 isfinite |      b       
----------+--------------
 f        | Positive INF
 f        | Negative INF
(2 rows)

set duckdb.force_execution = true;
-- DuckDB execution
SELECT * from t;
     a     |      b       
-----------+--------------
 infinity  | Positive INF
 -infinity | Negative INF
(2 rows)

SELECT isfinite(a),b FROM t;
 isfinite |      b       
----------+--------------
 f        | Positive INF
 f        | Negative INF
(2 rows)

-- Cleanup
set duckdb.force_execution = false;
DROP TABLE t;
SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00'::timestamp_s as timestamp_s $$);
         timestamp_s         
-----------------------------
 Mon Nov 24 00:00:00 4714 BC
(1 row)

SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59'::timestamp_s as timestamp_s $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The Timestamp value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59'::timestamp_s as timestamp_s $$);
        timestamp_s         
----------------------------
 Thu Dec 31 23:59:59 294246
(1 row)

SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00'::timestamp_s as timestamp_s $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The Timestamp value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00.000000'::timestamp as timestamp $$);
          timestamp          
-----------------------------
 Mon Nov 24 00:00:00 4714 BC
(1 row)

SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59.999999'::timestamp as timestamp $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The Timestamp value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59.999999'::timestamp as timestamp $$);
             timestamp             
-----------------------------------
 Thu Dec 31 23:59:59.999999 294246
(1 row)

SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00.000000'::timestamp as timestamp $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The Timestamp value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00.000'::timestamp_ms as timestamp_ms $$);
        timestamp_ms         
-----------------------------
 Mon Nov 24 00:00:00 4714 BC
(1 row)

SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59.999'::timestamp_ms as timestamp_ms $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The Timestamp value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59.999'::timestamp_ms as timestamp_ms $$);
          timestamp_ms          
--------------------------------
 Thu Dec 31 23:59:59.999 294246
(1 row)

SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00.000'::timestamp_ms as timestamp_ms $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The Timestamp value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
----------------------------------------
-- TimestampTz tests
----------------------------------------
-- Test +/- inf valuestz
CREATE TABLE t(a TIMESTAMPTZ, b TEXT);
INSERT INTO t VALUES('Infinity','Positive INF'), ('-Infinity','Negative INF');
-- PG Execution
SELECT * from t;
     a     |      b       
-----------+--------------
 infinity  | Positive INF
 -infinity | Negative INF
(2 rows)

SELECT isfinite(a), b FROM t;
 isfinite |      b       
----------+--------------
 f        | Positive INF
 f        | Negative INF
(2 rows)

set duckdb.force_execution = true;
-- DuckDB execution
SELECT * from t;
     a     |      b       
-----------+--------------
 infinity  | Positive INF
 -infinity | Negative INF
(2 rows)

SELECT isfinite(a), b FROM t;
 isfinite |      b       
----------+--------------
 f        | Positive INF
 f        | Negative INF
(2 rows)

-- Cleanup
set duckdb.force_execution = false;
DROP TABLE t;
SELECT * FROM duckdb.query($$ SELECT '4714-11-24 (BC) 00:00:00+00'::timestamptz as timestamptz $$);
           timestamptz           
---------------------------------
 Sun Nov 23 16:07:02 4714 LMT BC
(1 row)

SELECT * FROM duckdb.query($$ SELECT '4714-11-23 (BC) 23:59:59+00'::timestamptz as timestamptz $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The TimestampTz value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
SELECT * FROM duckdb.query($$ SELECT '294246-12-31 23:59:59+00'::timestamptz as timestamptz $$);
          timestamptz           
--------------------------------
 Thu Dec 31 15:59:59 294246 PST
(1 row)

SELECT * FROM duckdb.query($$ SELECT '294247-01-01 00:00:00+00'::timestamptz as timestamptz $$);  -- out of range
ERROR:  (PGDuckDB/Duckdb_ExecCustomScan_Cpp) Out of Range Error: The TimestampTz value should be between min and max value (4714-11-24 (BC) 00:00:00 <-> 294247-01-01 00:00:00)
